gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -415,9 +417,20 @@ public void markTaskCompletion() {
     if (this.countDownLatch.isPresent()) {
       this.countDownLatch.get().countDown();
     }
+    this.completed = true;
+
     this.taskState.setProp(ConfigurationKeys.TASK_RETRIES_KEY, this.retryCount.get());
   }
 
+  /**
+   * Return whether this {@link Task} is completed or not.
+   *
+   * @return {@code true} if this {@link Task} is completed or {@code false} otherwise
+   */
+  public boolean isCompleted() {
+    return this.completed;
```

Comment:
Can you use TaskState.COMMITTED/SUCCESSFUL instead? Also, you may want to clarify whether "completed" means succeeded, or either succeeded or failed.